### PR TITLE
Helix upgrade to Dynamo 2.7

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Place holder PR to inspect API changes and overall impact

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by QilongTang** [comment]
@DynamoDS/dynamo I have handled the conficts between master and this branch. Please give it some look and I think it is ready to go.

File: src/DynamoCoreWpf/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -40,3 +40,4 @@
 [assembly: InternalsVisibleTo("WorkspaceDependencyViewExtension")]
 [assembly: InternalsVisibleTo("DocumentationBrowserViewExtension")]
 [assembly: InternalsVisibleTo("SystemTestServices")]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Properties/AssemblyInfo.cs:43]: @QilongTang what is this?

Comment (QilongTang) [reply src/DynamoCoreWpf/Properties/AssemblyInfo.cs:43]: @mjkkirschner Brought in Moq dependency by https://github.com/DynamoDS/Dynamo/pull/10569. Details can be found here as well https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Properties/AssemblyInfo.cs#L24

Comment (mmisol) [reply src/DynamoCoreWpf/Properties/AssemblyInfo.cs:43]: That's correct. To be clear, that is the name of the assembly where mocks created using Moq are placed.

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 2]
```diff
@@ -81,6 +101,13 @@ private void RegisterButtonHandlers()
 
         private void RegisterViewEventHandlers()
         {
+            //var binding = new Binding("ActiveGeometry")
+            //{
+            //    Source = ViewModel,
+            //    Mode = BindingMode.OneWay,
+            //};
+            //watch_view.SetBinding(AttachedProperties.ActiveGeometryProperty, binding);
+
```
Comment (mmisol) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs:110]: NIT: We could clean up this commented out code.

File: src/VisualizationTests/HelixImageComparisonTests.cs [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+            Model.ExecuteCommand(command);
+            RunCurrentModel();
+            DynamoSelection.Instance.Selection.Add(pointOriginNode);
+            RenderCurrentViewAndCompare(MethodBase.GetCurrentMethod().Name);
+        }
+
+        #endregion
+
+    }
+}
```
Comment (mmisol) [inline src/VisualizationTests/HelixImageComparisonTests.cs:332]: NIT: Let's use `PascalCase` for naming all test methods, as that's the C# convention.

File: src/VisualizationTests/HelixWatch3dViewModelPrimitiveTests.cs [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+            Assert.False((mockManipulator.SceneNode.RenderCore as DynamoGeometryMeshCore).IsSpecialRenderPackageData);
+            AttachedProperties.SetIsSpecialRenderPackage(mockManipulator, true);
+            //assert that setting this attached property updated the meshcore data.
+            Assert.IsTrue((mockManipulator.SceneNode.RenderCore as DynamoGeometryMeshCore).IsSpecialRenderPackageData);
+          
+        }
+        #endregion
+
+    }
+}
```
Comment (mmisol) [inline src/VisualizationTests/HelixWatch3dViewModelPrimitiveTests.cs:533]: NIT: Let's use `PascalCase` (that means no `_` separator) in method names.

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 5]
```diff
@@ -54,17 +51,14 @@
       <Private>True</Private>
       <HintPath>..\packages\Greg.1.3.7408.23353\lib\net47\Greg.dll</HintPath>
     </Reference>
-    <Reference Include="HelixToolkit, Version=2015.1.629.0, Culture=neutral, PublicKeyToken=52aa3500039caf0d, processorArchitecture=MSIL">
-      <HintPath>..\packages\HelixToolkit.2015.1.629\lib\portable-net4+sl4+wp71+win8\HelixToolkit.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="HelixToolkit, Version=2.9.0.0, Culture=neutral, PublicKeyToken=52aa3500039caf0d, processorArchitecture=MSIL">
+      <HintPath>..\packages\HelixToolkit.2.9.0\lib\netstandard1.1\HelixToolkit.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (mjkkirschner) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:55]: @QilongTang I didn't notice this was .net standard, does this have the same issue as the pythonNet net standard stuff and bring in a ton of type forwarding system binaries? We may want to try updating dynamo to 4.7.2 ASAP.

File: test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj [hunk 6]
```diff
@@ -89,36 +88,28 @@
...
+      <Private>True</Private>
     </Reference>
-    <Reference Include="SharpDX.Direct2D1, Version=2.6.3.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1, processorArchitecture=MSIL" />
-    <Reference Include="SharpDX.Direct3D11, Version=2.6.3.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1, processorArchitecture=MSIL">
-      <HintPath>$(SharpDXPackageBinDir)\SharpDX.Direct3D11.dll</HintPath>
+    <Reference Include="SharpDX.D3DCompiler, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1, processorArchitecture=MSIL">
+      <HintPath>..\..\src\packages\SharpDX.D3DCompiler.4.2.0\lib\net45\SharpDX.D3DCompiler.dll</HintPath>
     </Reference>
-    <Reference Include="SharpDX.Direct3D11.Effects, Version=2.6.3.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1, processorArchitecture=MSIL">
-      <HintPath>$(SharpDXPackageBinDir)\SharpDX.Direct3D11.Effects.dll</HintPath>
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: these effects binaries will likely be removed from the bin folder after this PR - may should consider keeping them around I guess to support patch installers?

Comment (mmisol) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: I see that dll still being referenced in Watch3DNodeModelsWpf.csproj. Should it not be there anymore?

Comment (mjkkirschner) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: we should definitely get rid of it if we can.
@QilongTang do you want to try to take care of it here or should we file another task to look at that reference?

Comment (QilongTang) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: @mjkkirschner If we get rid of it, do we still need an empty binary with same name around?

Comment (mjkkirschner) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: yes, I think so, but I think what is more important is to understand if we actually still *need* that reference for some reason.

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 7]
```diff
@@ -48,6 +48,7 @@
 [assembly: AssemblyVersion("2.7.0.8085")]
 
 
+
```
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:51]: Will update this file

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs:110]: Will remove

File: src/VisualizationTests/HelixImageComparisonTests.cs (hunk 3 above)
Comment (QilongTang) [reply src/VisualizationTests/HelixImageComparisonTests.cs:332]: Updated

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:55]: I dont see a tons of added system binaries in the built folder

File: test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj (hunk 6 above)
Comment (QilongTang) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: @mjkkirschner Then should be revert this change for now?

File: src/VisualizationTests/HelixWatch3dViewModelPrimitiveTests.cs (hunk 4 above)
Comment (QilongTang) [reply src/VisualizationTests/HelixWatch3dViewModelPrimitiveTests.cs:533]: Updated

File: test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj (hunk 6 above)
Comment (mjkkirschner) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: no, don't revert, what I meant was let's try removing the reference from `Watch3DNodeModelsWpf` and see if the build works.

Comment (QilongTang) [reply test/DynamoCoreWpfTests/DynamoCoreWpfTests.csproj:109]: Will remove

**Comment by mjkkirschner** [comment]
@QilongTang changing the test names in the image comparison tests will change the names of the output files that get generated - that might cause some failure in the test, or the test not to really do a proper comparison, please double check that if you want to proceed with updating the test method names.

**Comment by QilongTang** [comment]
@mjkkirschner The self serve pass and DisplayByGeometry tests pass in WpfVisualization tests, any other ones worth checking here?

**Comment by QilongTang** [comment]
@DynamoDS/dynamo Merging this. Will include the helix newer version update in a different PR.

**Pull request merged**